packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkNativeInitializationTest.kt

```diff
@@ -0,0 +1,168 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import android.content.pm.PackageInfo
+import com.datadog.tools.unit.GenericAssert.Companion.assertThat
+import com.datadog.tools.unit.forge.BaseConfigurator
+import com.facebook.react.bridge.ReactApplicationContext
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.net.InetSocketAddress
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Answers
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.isNull
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = BaseConfigurator::class)
+internal class DdSdkNativeInitializationTest {
+
+    lateinit var testedNativeInitialization: DdSdkNativeInitialization
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    lateinit var mockContext: ReactApplicationContext
+
+    @Forgery
+    lateinit var mockPackageInfo: PackageInfo
+
+    @Mock
+    lateinit var mockDatadog: DatadogWrapper
+
+    @Mock
+    lateinit var mockJSONFileReader: JSONFileReader
+
+    @BeforeEach
+    fun `set up`() {
+        whenever(mockContext.packageName) doReturn "packageName"
+        whenever(
+            mockContext.packageManager.getPackageInfo(
+                "packageName",
+                0
+            )
+        ) doReturn mockPackageInfo
+
+        testedNativeInitialization = DdSdkNativeInitialization(
+            mockContext,
+            mockDatadog,
+            mockJSONFileReader
+        )
+    }
+
+    // region getConfigurationFromJSONFile
+
+    @Test
+    fun `𝕄 parse configuration file 𝕎 getConfigurationFromJSONFile() {file with all data}`() {
+        whenever(
```

Comment:
let's split into `Given/When/Then` sections and add new lines for the readability (applies also to the tests below)